File: backend/app/Models/Payment.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\SoftDeletes;

class Payment extends Model
{
    use HasFactory, SoftDeletes;

    protected $fillable = [
        'payment_number', 'ride_id', 'user_id', 'payment_method_id',
        'payment_type', 'amount', 'platform_fee', 'driver_amount',
        'status', 'gateway', 'gateway_transaction_id', 'gateway_payment_id',
        'gateway_response', 'pix_qr_code', 'pix_qr_code_base64',
        'pix_transaction_id', 'pix_expires_at', 'card_brand',
        'card_last_four', 'installments', 'refund_amount',
        'refund_reason', 'refunded_at', 'processed_at',
        'failed_at', 'failure_reason',
    ];

    protected function casts(): array
    {
        return [
            'amount' => 'decimal:2',
            'platform_fee' => 'decimal:2',
            'driver_amount' => 'decimal:2',
            'refund_amount' => 'decimal:2',
            'pix_expires_at' => 'datetime',
            'refunded_at' => 'datetime',
            'processed_at' => 'datetime',
            'failed_at' => 'datetime',
        ];
    }

    public function ride()
    {
        return $this->belongsTo(Ride::class);
    }

    public function user()
    {
        return $this->belongsTo(User::class);
    }

    public function paymentMethod()
    {
        return $this->belongsTo(PaymentMethod::class);
    }

    public function isPending(): bool
    {
        return $this->status === 'pending';
    }

    public function isCompleted(): bool
    {
        return $this->status === 'completed';
    }

    public function isFailed(): bool
    {
        return $this->status === 'failed';
    }
}

File: packages/mobi_core/lib/src/models/payment.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:equatable/equatable.dart';

part 'payment.g.dart';

@JsonSerializable()
class Payment extends Equatable {
  final int id;
  @JsonKey(name: 'payment_number')
  final String paymentNumber;
  @JsonKey(name: 'ride_id')
  final int rideId;
  @JsonKey(name: 'user_id')
  final int userId;
  @JsonKey(name: 'payment_type')
  final String paymentType;
  final double amount;
  @JsonKey(name: 'platform_fee')
  final double? platformFee;
  @JsonKey(name: 'driver_amount')
  final double? driverAmount;
  final String status;
  final String gateway;
  @JsonKey(name: 'gateway_payment_id')
  final String? gatewayPaymentId;
  @JsonKey(name: 'pix_qr_code')
  final String? pixQrCode;
  @JsonKey(name: 'pix_qr_code_base64')
  final String? pixQrCodeBase64;
  @JsonKey(name: 'card_brand')
  final String? cardBrand;
  @JsonKey(name: 'card_last_four')
  final String? cardLastFour;
  final int? installments;
  @JsonKey(name: 'created_at')
  final DateTime? createdAt;
  @JsonKey(name: 'processed_at')
  final DateTime? processedAt;

  const Payment({
    required this.id,
    required this.paymentNumber,
    required this.rideId,
    required this.userId,
    required this.paymentType,
    required this.amount,
    this.platformFee,
    this.driverAmount,
    required this.status,
    required this.gateway,
    this.gatewayPaymentId,
    this.pixQrCode,
    this.pixQrCodeBase64,
    this.cardBrand,
    this.cardLastFour,
    this.installments,
    this.createdAt,
    this.processedAt,
  });

  factory Payment.fromJson(Map<String, dynamic> json) => _$PaymentFromJson(json);
  Map<String, dynamic> toJson() => _$PaymentToJson(this);

  bool get isPending => status == 'pending';
  bool get isProcessing => status == 'processing';
  bool get isCompleted => status == 'completed';
  bool get isFailed => status == 'failed';

  bool get isPix => paymentType == 'pix';
  bool get isCreditCard => paymentType == 'credit_card';
  bool get isDebitCard => paymentType == 'debit_card';
  bool get isCash => paymentType == 'cash';

  @override
  List<Object?> get props => [
        id,
        paymentNumber,
        rideId,
        userId,
        paymentType,
        amount,
        platformFee,
        driverAmount,
        status,
        gateway,
        gatewayPaymentId,
        pixQrCode,
        pixQrCodeBase64,
        cardBrand,
        cardLastFour,
        installments,
        createdAt,
        processedAt,
      ];
}
